LibGit2Sharp/Core/Proxy.cs

```diff
@@ -2167,6 +2167,17 @@ public static unsafe void git_remote_connect(RemoteHandle remote, GitDirection d
             }
         }
 
+        public static unsafe string git_remote_default_branch(RemoteHandle remote)
+        {
+            using (var buf = new GitBuf())
+            {
+                int res = NativeMethods.git_remote_default_branch(buf, remote);
+                Ensure.ZeroResult(res);
+
+                return LaxUtf8Marshaler.FromNative(buf.ptr);
```

Comment:
Its probably not our place to fix but how does this handle a branch using unicode 16 like say something with a Cyrillic 
 branch name?

Reply:
I don't think it will ever be using UTF-16 tbh. It appears that the encoding for the native library is UTF-8. Utf8Marshaler from this library uses UTF8Encoding class so it's not a simple ANSI. It's also the same marshaler that's used to send branch names to libgit2. I believe it would work fine.